### DISCUSSION TITLE
How to update vue-tsc? (Obsolete)

### DISCUSSION THREAD

**Discussion by johnsoncodehk** [vuejs/language-tools#402] [Show and tell]
> How to update vscode-vue-languageservice in vue-tsc?

**npm**:
~~delete package-lock.json and reinstall it.~~
run `$ npm uninstall vue-tsc && npm cache clean --force && npm install vue-tsc` (thanks to @henribru!)

**yarn**:
delete yarn.lock and reinstall it.

**pnpm**:
run `$ pnpm up --latest vue-tsc` (thanks to @KyllianV!)

> How to roll back to specific version?

If use yarn, you can config `resolutions` in package.json to roll back to a specific version:

```json
{
  ...
  "resolutions": {
    "vscode-vue-languageservice": "0.27.xx"
  }
}
```

:warning: If template type-check has problems after the updated, probably related to #322.

**Comment by zhixiangyao** [comment]
Thx~

**Comment by henribru** [comment]
This can actually be accomplished without having to blow away your lock file. If you run `npm uninstall vue-tsc` and then `npm install vue-tsc` you'll get the same version of the transitive dependencies, but this just seems to be because npm caches the resolutions. If you run `npm cache clean --force` inbetween the uninstall and reinstall you'll get the newest versions.

**Reply by jsonkody** [reply]
Thx :D .. that works actually better than deleting package-lock

**Reply by henribru** [reply]
@johnsoncodehk Sorry, I think you missed a step when adding my solution to your post (at least, for me it doesn't work without the middle step). The cache should be cleared inbetween, so `npm uninstall vue-tsc && npm cache clean --force && npm install vue-tsc`

**Reply by johnsoncodehk** [reply]
@henribru ah my bad, thanks again!

**Comment by tjx666** [comment]
pnpm doesn't work with`pnpm up --latest vue-tsc`.

![image](https://user-images.githubusercontent.com/41773861/135714144-b60e466a-833e-48f0-b927-8068333ef42c.png)

**Reply by tjx666** [reply]
I finally manually add dev dependency of `vscode-vue-languageservice@0.27.27`

**Reply by johnsoncodehk** [reply]
cc @KyllianV

**Reply by KyllianV** [reply]
The `pnpm up --latest vue-tsc` works well though. Your `--filter` is probably the cause of your problem. Try without it, but anyway do not add `vscode-vue-languageservice` to your dependencies it's useless.

![image](https://user-images.githubusercontent.com/15817431/136253479-c50c4435-619b-4084-b9f4-0bad193d3981.png)

**Reply by tjx666** [reply]
@KyllianV My Project is a project which need to enable `shamefully-hoist=true` in .npmrc

**Reply by KyllianV** [reply]
I don't think it's related to the hoisting, if `vue-tsc` is in your `package.json` you should be able to update it with this command. `vscode-vue-languageservice` is already a dependency of `vue-tsc` so if you are not using it yourself in your code, you shouldn't need to add it in your `package.json` file.

Did you try without the `--filter` ?

**Reply by tjx666** [reply]
@KyllianV 
vue-tsc is a devdependencies in one package, not workspace dependencies.

![image](https://user-images.githubusercontent.com/41773861/136417946-6195d5ec-68d9-4078-b3f2-ade8b6c656db.png)

**Reply by KyllianV** [reply]
Ok well I meant to run the command within the right package too but maybe it will not work. I'm not an expert you should try all the options listed in the documentation of pnpm or submit an issue in their repo. Your issue is more a misunderstanding of pnpm with your architecture than a problem with `vscode-vue-languageservice`. In my project (which is a monorepo with multiple pnpm workspaces) I was able to update the package with this command. I'm sorry I'm not able to help you more with this :/

**Reply by johnsoncodehk** [reply]
@KyllianV thanks for you helping! After v0.28, vue-tsc is fixed version and follow to volar, you just need to use `"vue-tsc": "0.28.x"` in package.json. So I think it's not a problem in future.
